### PR TITLE
Adapt upgrade guide for katello

### PR DESCRIPTION
**Pull request by jlsherrill**
Cherry-pick into:

* [ ] Foreman 2.3 (Satellite 6.9)
* [ ] Foreman 2.1 (Satellite 6.8)

### CODE REVIEW

**Comment by jlsherrill** [comment]
This is a work in progress for the upgrade docs, opening for early feedback

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (jlsherrill) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:46]: i guess this isn't actually  needed since this guide won't exist for 2.3.  TODO: DELETE THIS

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (lzap) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:17]: TODO reminder.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (lzap) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:50]: Nice solution :)

**Comment by lzap** [comment]
This looks good to me, someone from Katello team should probably review tho.

File: guides/doc-Upgrading_and_Updating/topics/cloning_satellite_server.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/cloning_satellite_server.adoc:199]:
```suggestion
. Update your network configuration, for example, DNS, to match the target server’s IP address with its new host name. The `satellite-clone` tool changes the hostname to the source server's hostname. If you want to change the hostname to something different, you can use the `satellite-change-hostname` tool. For more information, see {AdministeringDocURL}sect-Red_Hat_Satellite-Administering_Red_Hat_Satellite-Renaming_a_Server[Renaming a {Project} or {SmartProxyServer}] in _Administrating {ProjectName}_.
```

File: guides/doc-Upgrading_and_Updating/topics/cloning_satellite_server.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/cloning_satellite_server.adoc:201]: This will fail the link validator as we don't have an upstream version of this link and the BASEURL will cause the linkchecker to validate the docs.theforeman.org/configuring_virtual_machine_subscriptions_in_red_hat_satellite/index.html etc etc
I removed the baseurl for that purpose.

File: guides/common/attributes.adoc
Comment (melcorr) [inline guides/common/attributes.adoc:47]: Oh oh oh. I love this!

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (melcorr) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:17]: What do you mean by fix this? Link to the this doc ? https://docs.theforeman.org/nightly/Administering_Red_Hat_Satellite/index-foreman-el.html#backing-up-satellite-server-and-capsule-server ? What else is needed?

File: guides/common/attributes.adoc
Comment (melcorr) [inline guides/common/attributes.adoc:139]: Can you explain what this does that wasn't possible beforehand?

**Review by melcorr — COMMENTED** [review]
All looks good. A question here or there/

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ehelms) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:93]: Can drop the indention here to match up with other declarations

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ehelms) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:107]: There is a macro for this, I think `{foreman-installer}`

**Comment by melcorr** [comment]
Hey @jlsherrill - is there anything I've blocked you with here? 
@lzap  is proposing a large change with #460 that will impact your proposed attributes also. 
I'm assuming this will need to be finished soon enough?

**Comment by jlsherrill** [comment]
Sorry, about that!  i will update this today

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (jlsherrill) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:17]: yep, adjusting it to the new url

File: guides/common/attributes.adoc
Comment (jlsherrill) [reply guides/common/attributes.adoc:139]: it made it hard to refer to the previous version of the docs, like how to backup the previous version

**Comment by jlsherrill** [comment]
i think i've addressed all the comments, i think i'm getting 404s because i've got links referring to the '2.4' docs, but they are currently at /nightly/.  Will work on that!

**Comment by jlsherrill** [comment]
🟢

**Comment by jlsherrill** [comment]
I just realized i could do this! 

🔴 | 
-------|
🟡  |
🟢  |

**Comment by lzap** [comment]
Feel free to merge what comes first, I can rebase my PR too.

I am good with this one, let's merge?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:98]: Install here is weird. It should already be present, so upgrade?
```suggestion
# yum update -y centos-release-scl-rh
```

Also, this is a CentOS users part. RHEL users should enable the right channel instead.

Thinking more about it, this should already be covered in the `yum update -y` just before this so:
```suggestion
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:105]: I think you don't need to clean everything. Perhaps
```suggestion
# yum clean expire-cache
```

Or if that's insufficient:
```suggestion
# yum clean metadata
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:96]: I know @evgeni is working on moving this. Perhaps https://yum.theforeman.org/katello/ should be used already?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:97]: I would like to suggest to swap the order: first `foreman-release.rpm` and then update katello. Mostly for a future simplification: I think that we should make a `foreman-release-katello` in the Foreman repository. Then the installation instructions could become:
```
yum install -y https://yum.theforeman.org/releases/{ProjectVersion}/el7/x86_64/foreman-release.rpm
yum install -y foreman-release-katello centos-release-scl-rh
```
(name could also be `katello-release`, not important for the example)

The benefit is that there is no (reasonable) way for a user to install a mismatched version.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:121]: I don't understand this suggestion. Why do you need to stop all services and is just `reboot` insufficient? Is that always required? If so, IMHO it's a major limitation that we must think about.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:133]: IMHO this is unsupported and we should not document it. https://theforeman.org/plugins/katello/nightly/installation/index.html#multiple-subnets-and-domains documents how to handle this in the installer. Perhaps out of scope for this PR.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:140]: When would this happen? I don't know OpenSCAP but is this something seeding should take care of? If so, IMHO it can indicate a bug that we have to document this.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:39]: While you're at it, could you hide the above section behind a katello/satellite definition? Vanilla Foreman doesn't need `rh-postgresql12-postgresql-evr`.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:24]: Also noted below, but personally I think we should not encourage this.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (ekohl) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:17]: While the whole document is very focused on EL, we may want to create an issue to write similar instructions for Debian.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (evgeni) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:96]: Yes please. 3.17, 3.18 and 4.0 is installable from yum.tfm today.

**Comment by lzap** [comment]
Hello,

we have merged a big cleanup in `attributes.adoc` file, it is now split into several files:

* [attributes.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes.adoc): version definitions and includes for other attribute files.
* [attributes-base.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-base.adoc): base attributes common for all builds.
* [attributes-foreman-el.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-foreman-el.adoc): base overrides for foreman-el build.
* [attributes-foreman-deb.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-foreman-deb.adoc): base overrides for foreman-deb build.
* [attributes-katello.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-katello.adoc): base overrides for katello build.
* [attributes-satellite.adoc](https://github.com/theforeman/foreman-documentation/blob/master/guides/common/attributes-satellite.adoc): base overrides for satellite build.

Please resolve the conflicts and update the attributes accordingly, sorry for this but we believe this will ultimately lead to cleaner attributes. I wish you nice Easter!

**Comment by lzap** [comment]
I am about to branch off 2.4 series, what is the status please? I can hold off the branching process if this is something that should be in 2.4/4.0. Let me know, thanks.

**Comment by jlsherrill** [comment]
Apologies, i will work on this tonight :)

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (jlsherrill) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:121]: i agree, it doesn't make much sense.  Let me change to just 'reboot the system'.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (jlsherrill) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:140]: i'm deleting this, as i don't understand why its needed

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (jlsherrill) [reply guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:133]: While i agree, i think maybe out of scope for this pr, maybe we should open an issue to incorporate https://theforeman.org/plugins/katello/nightly/installation/index.html#multiple-subnets-and-domains

**Comment by jlsherrill** [comment]
@ekohl i think i addressed all your comments other than those you mentioned might be out of scope for this PR

**Comment by jlsherrill** [comment]
Thanks @lzap for the patience!

**Comment by lzap** [comment]
Can I get re-reviews please? Thank you all!

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:136]: This isn't your fault but 
```suggestion
[options="nowrap" subs="attributes"]
```

**Review by melcorr — CHANGES_REQUESTED** [review]
Hi @jlsherrill 

Can you do a local test of this? I think that the master.adoc file might need some tweaking. The nav menu that I see only has the following sections:

    1. Upgrade Overview
        1.1. Prerequisites
        1.2. Post-Upgrade Tasks
        1.3. Reclaiming MongoDB Space
        1.4. Reclaiming PostgreSQL Space
        1.5. Updating Satellite Server, Capsule Server, and Content Hosts

I cannot find a lot of the content that you've edited here. I know you've suppressed some Satellite-only content but this seems like more than this.

File: guides/doc-Upgrading_and_Updating/master.adoc
Comment (jlsherrill) [reply guides/doc-Upgrading_and_Updating/master.adoc:44]: i think this was a rebase error, good catch!

**Comment by jlsherrill** [comment]
@melcorr i did not test it after rebasing, apparently i should have!   Fixed, as well as removed some mongo docs.  

I noticed that the smart proxy upgrade part also needs a it of work, but i propose to work on this as a followup

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:103]:
```suggestion
. Run the installer
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:88]: Is installer-log-file defined anywhere? When I build this as Katello, I get the following: 

If you lose connection to the command shell where the upgrade command is running, you can see the logged messages in the /var/log/foreman-installer/satellite.log file to check if the process completed successfully.

File: guides/doc-Upgrading_and_Updating/topics/introduction_updating_satellite.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/introduction_updating_satellite.adoc:1]: The title for this is rendering as: 

Updating Satellite Server, Capsule Server, and Content Hosts

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:44]:
```suggestion
* Back up your {ProjectServer} and all {SmartProxyServer}s. For more information, see {AdministeringDocURL}#backing-up-satellite-server-and-capsule-server[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProductVersionPrevious}_ guide.
```
Without the text in [] this just prints a hyperlink. It's good practice to recommend the previous version of the guide because the backup and restore docs might shift from version to version.

File: guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_process_overview.adoc:72]:
```suggestion
ifdef::satellite[]
. Clone your existing {ProjectServer}s. For more information, see xref:cloning_satellite_server[].
endif::[]
```

This is broken as there are no cloning instructions included in the Katello upgrade guide?

File: guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_a_connected_satellite_server.adoc:96]:
```suggestion
Clean the yum cache and update the required packages
```

File: guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc
Comment (melcorr) [inline guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc:11]: The xref here doesn't work because this section has been filtered out of the Katello guide. Do we need it? If so, add it in, if not, wrap it in a Satellite ifdef

**Review by melcorr — CHANGES_REQUESTED** [review]
Hi Justin, 
A few final things!

**Comment by lzap** [comment]
Once you merge this @melcorr please ping me so I can rebase and finalize the branching process.

**Comment by melcorr** [comment]
@jlsherrill if you do not have time, please let me know, I'll merge then apply these suggestions myself. I know this has been a PITA

**Comment by jlsherrill** [comment]
@melcorr if you don't mind, that would be amazing, I'm head deep in some fixes for 6.9.1 :)

**Comment by melcorr** [comment]
@lzap  please don't branch until later when I finish this !!

**Comment by melcorr** [comment]
@lzap i completed this in #488 - are we rushing a bit with the branching? Katello 4.0 is the first time we will be official. No one is expecting us to have 2.4. If we merge now, we have a lot of cherry-picking. What do you think?

**Pull request merged**
